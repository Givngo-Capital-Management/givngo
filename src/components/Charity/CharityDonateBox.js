import React from 'react';
import { Button, Grid } from 'semantic-ui-react';
import DonateModal from '../Stripe/DonateModal';
import styles from './CharityDonateBox.module.scss';

export default function CharityDonateBox(props) {
  const { charity } = props

  const donateButton = (
    <Button className="donate ui right floated" color="orange">
      Donate
    </Button>
  )

  return (
    <div className={styles.box}>
      <div className={styles.header}>
        <p>Donate now</p>
      </div>
      <div className={styles.content}>
        <Grid>
          <div className="equal width row">
            <div className="ten wide column">
              <h3 className="ui header">
                <div className="content">
                  {`Donate to ${charity.name}`}
                </div>
              </h3>
            </div>
            <div className="six wide column">
              <DonateModal charity={charity} button={donateButton}/>
            </div>
          </div>
        </Grid>

        <h3>{`Total raised: $${charity.total_raised || 0}`}</h3>
      </div>
    </div>
  );
}
