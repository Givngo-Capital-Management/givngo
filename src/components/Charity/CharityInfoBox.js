import React from 'react';
import { Image, Button } from 'semantic-ui-react';
import styles from './CharityInfoBox.module.scss';

export default function CharityInfoBox(props) {
  const { charity } = props
  return (
    <div className={styles.box}>
      <Image
        className={styles.image}
        src={charity.logo}
        circular
      />
      <div className={styles.info}>
        <div className="charity-name">{charity.name}</div>
        <span>80k followers</span>
      </div>
      <Button className={styles.follow} color="orange">
        Follow
      </Button>
      <div className={styles.description}>
        <p>{charity.description}</p>
      </div>
    </div>
  );
}
