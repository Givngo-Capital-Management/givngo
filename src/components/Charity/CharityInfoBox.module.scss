.box {
  display: grid;
  grid: auto auto / 58px auto max-content;
  align-items: center;
  grid-row-gap: 16px;

  .image {
    grid-row: 1 / 2;
    grid-column: 1 / 2;
  }

  .info {
    grid-row: 1 / 2;
    grid-column: 2 / 3;

    .name {
      font-weight: 600;
      cursor: pointer;
    }
    .date {
      font-size: 13px;
      color: #707070;
    }
  }

  .follow {
    grid-row: 1 / 2;
    grid-column: 3 / 4;
  }

  .description {
    max-width: 615px;
    grid-row: 2 / 3;
    grid-column: 2 / 3;

    button {
      margin: 8px 0;
    }
  }
}