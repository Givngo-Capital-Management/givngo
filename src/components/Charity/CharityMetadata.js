import React from 'react';
import { Button, Divider, Icon } from 'semantic-ui-react';
import styles from './CharityMetadata.module.scss';
import moment from 'moment';

export default function CharityMetadata(props) {
  const { charity } = props;
  const date = moment.utc(charity.video_date * 1000).format('MMMM DD, YYYY').toString()
  return (
    <div className={styles.metadata}>
      <h3>{charity.video_title}</h3>
      <div className={styles.stats}>
        <span>{date}</span>
        <div className={styles.actions}>
          <Button basic icon labelPosition="left">
            <Icon name="share" />
            Share
          </Button>
        </div>
      </div>
      <Divider />
    </div>
  );
}
