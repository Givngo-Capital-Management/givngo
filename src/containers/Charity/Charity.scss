.charity-detail-grid {
  display: grid;
  grid-template: auto auto auto 1fr / minmax(0, 1280px) 410px;
  justify-content: center;
  padding-top: 24px;
  column-gap: 24px;
  grid-row-gap: 8px;

  .video {
    grid-column: 1 / 2;
    grid-row: 1 / 2
  }
  .metadata {
    grid-column: 1 / 2;
    grid-row: 2 / 3
  }
  .charity-info-box {
    grid-column: 1 / 2;
    grid-row: 3 / 4;
  }
  .charity-detail-sidebar {
    grid-column: 2 / 2;
    grid-row: 1 / span 4 ;
  }
}

// 1280px (max-width of video column) + 410px (width of side bar) + 3 * 24px (empty space on the left, right and between the two columns)
@media(max-width: 1762px) {
  .watch-grid {
    padding-left: 24px;
    padding-right: 24px;
  }
}