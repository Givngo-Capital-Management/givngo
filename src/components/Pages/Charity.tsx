import React from 'react';
import {Grid} from 'semantic-ui-react';

import AppLayout from '../AppLayout';
import Video from '../Charity/Video';
import RelatedCharities from '../Charity/RelatedCharities';
import CharityMetadata from '../Charity/CharityMetadata';
import CharityInfoBox from '../Charity/CharityInfoBox';
import CharityDonateBox from '../Charity/CharityDonateBox';

import 'firebase/auth';

const Charity = (props: any) => {
  return (
    <AppLayout>
      <Video id="374481280" title="breast friends" />
      <div style={{ paddingLeft: 20, paddingRight: 20 }}>
        <Grid columns={2}>
          <Grid.Column width={10}>
            <CharityMetadata />
            <CharityInfoBox />
          </Grid.Column>
          <Grid.Column width={6} floated="right">
            <div>
              <CharityDonateBox />
              <RelatedCharities />
            </div>
          </Grid.Column>
        </Grid>
      </div>
    </AppLayout>
  )
};

export default Charity;