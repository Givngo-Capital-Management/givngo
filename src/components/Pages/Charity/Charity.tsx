import React, { useState, useEffect } from 'react';
import {Grid} from 'semantic-ui-react';

import AppLayout from '../../AppLayout';
import Video from '../../Charity/Video';
import RelatedCharities from '../../Charity/RelatedCharities';
import CharityMetadata from '../../Charity/CharityMetadata';
import CharityInfoBox from '../../Charity/CharityInfoBox';
import CharityDonateBox from '../../Charity/CharityDonateBox';

import firebase from 'firebase/app';
import 'firebase/database';
import {Charity} from '../../../types';
import { useRouter } from 'next/router';

const CharityPage = (props: any) => {
  const [charity, setCharity] = useState<Charity>();
  const router = useRouter();

  const { id } = router.query;

  useEffect(() => {
    const listener = firebase.database()
      .ref(`charities/${id}`)
      .on('value', (snapshot) => {
        if (snapshot) {
          setCharity(snapshot.val());
        }
      });
    return listener(null);
  }, []);

  return (
    <AppLayout>
      <Video id={charity?.video_id} title={charity?.video_title} />
      <div style={{ paddingLeft: 20, paddingRight: 20 }}>
        <Grid columns={2}>
          <Grid.Column width={10}>
            <CharityMetadata charity={charity ? charity : {}} />
            <CharityInfoBox charity={charity ? charity : {}} />
          </Grid.Column>
          <Grid.Column width={6} floated="right">
            <div>
              <CharityDonateBox charity={charity ? charity : {}}/>
              <RelatedCharities />
            </div>
          </Grid.Column>
        </Grid>
      </div>
    </AppLayout>
  )
};

export default CharityPage;