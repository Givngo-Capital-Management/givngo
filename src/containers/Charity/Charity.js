import React from 'react';
import Video from '../../components/Video/Video';
import RelatedCharities from '../../components/RelatedCharities/RelatedCharities';
import './Charity.scss';
import CharityMetadata from '../../components/CharityMetadata/CharityMetadata';
import CharityInfoBox from '../../components/CharityInfoBox/CharityInfoBox';

export default function Charity() {
  return (
    <div className="charity-detail-grid">
      <Video className="video" id="396392696" title="breast friends" />
      <CharityMetadata className="metadata" />
      <CharityInfoBox className="charity-info-box" />
      <RelatedCharities className="related-charities" />
    </div>
  );
}
