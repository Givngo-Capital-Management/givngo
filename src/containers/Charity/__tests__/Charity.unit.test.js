import React from 'react';
import { shallow } from 'enzyme';
import Charity from '../Charity';

describe('Charity', () => {
  test('renders', () => {
    const wrapper = shallow(<Charity />);
    expect(wrapper).toMatchSnapshot();
  });
});
