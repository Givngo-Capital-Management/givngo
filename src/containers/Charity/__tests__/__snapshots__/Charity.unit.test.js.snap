// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charity renders 1`] = `
<div
  className="charity-detail-grid"
>
  <Video
    className="video"
    id="396392696"
    title="breast friends"
  />
  <CharityMetadata
    className="metadata"
  />
  <CharityInfoBox
    className="charity-info-box"
  />
  <RelatedCharities
    className="related-charities"
  />
</div>
`;
